File: 3.SpringBoot/src/main/java/com/practice/studio/entity/Recording.java
package com.practice.studio.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "records")
public class Recording {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne()
    @JoinColumn(name = "artist_id")
    private Artist artist;

    @ManyToOne()
    @JoinColumn(name = "employee_id")
    private Employee employee;

    @ManyToOne()
    @JoinColumn(name = "song_id")
    private Song song;

    @Column(name = "record_date")
    private LocalDate date;
}

File: 4.Angular/src/app/model/recording.ts
export class Artist {
    id!: number;
}

export class Employee {
    id!: number;
}

export class Song {
    id!: number;
}

export class Recording {
    id!: number;
    artist!: Artist;
    employee!: Employee;
    song!: Song;
    date!: string;
  }
  